src/AnimationManager.ts

```diff
@@ -1,377 +1,468 @@
-
-					// Subtract distance from distance already moved.
-					// Recalculate the remaining distance.
-					// Fix the bouncing phenomenon by changing the range.
-					const circulatedPos = getCirculatedPos(nextPos, options.range, options.circular as boolean[]);
-					if (nextPos !== circulatedPos) {
-						// circular
-						const rangeOffset = directions[key] * (options.range[1] - options.range[0]);
-
-						animateParam.destPos[key] -= rangeOffset;
-						prevPos[key] -= rangeOffset;
-					}
-					return circulatedPos;
-				});
-				const isCanceled = !self.em.triggerChange(toPos, false, prevPos);
-
-				prevPos = toPos;
-				self._prevEasingPer = easingPer;
-				if (easingPer >= 1) {
-					animateParam.destPos = self.getFinalPos(animateParam.destPos, originalIntendedPos);
-
-					if (!equal(animateParam.destPos, self.axm.get(Object.keys(animateParam.destPos)))) {
-						self.em.triggerChange(animateParam.destPos, true, prevPos);
-					}
-					complete();
-					return;
-				} else if (isCanceled) {
-					self.finish(false);
-				} else {
-					// animationEnd
-					self._raf = requestAnimationFrame(loop);
-				}
-			})();
-		} else {
-			this.em.triggerChange(param.destPos, true);
-			complete();
-		}
-	}
-
-	/**
-	 * Get estimated final value.
-	 *
-	 * If destPos is within the 'error range' of the original intended position, the initial intended position is returned.
-	 *   - eg. original intended pos: 100, destPos: 100.0000000004 ==> return 100;
-	 * If dest Pos is outside the 'range of error' compared to the originally intended pos, it is returned rounded based on the originally intended pos.
-	 *   - eg. original intended pos: 100.123 destPos: 50.12345 => return 50.123
-	 *
-	 * @param originalIntendedPos
-	 * @param destPos
-	 */
-	private getFinalPos(destPos: ObjectInterface<number>, originalIntendedPos: ObjectInterface<number>): Axis {
-		// compare destPos and originalIntendedPos
-		const ERROR_LIMIT = 0.000001;
-		const finalPos = map(destPos, (value, key) => {
-			if (value >= originalIntendedPos[key] - ERROR_LIMIT && value <= originalIntendedPos[key] + ERROR_LIMIT) {
-				// In error range, return original intended
-				return originalIntendedPos[key];
-			} else {
-				// Out of error range, return rounded pos.
-				const roundUnit = this.getRoundUnit(value, key);
-				const result = roundNumber(value, roundUnit);
-				return result;
-			}
-		});
-		return finalPos;
-	}
-
-	private getRoundUnit(val: number, key: string): number {
-		const roundUnit = this._options.round; // manual mode
-		let minRoundUnit = null; // auto mode
-
-		// auto mode
-		if (!roundUnit) {
-			// Get minimum round unit
-			const options = this.axm.getAxisOptions(key);
-			minRoundUnit = inversePow(Math.max(
-				getDecimalPlace(options.range[0]),
-				getDecimalPlace(options.range[1]),
-				getDecimalPlace(val)));
-		}
-
-		return minRoundUnit || roundUnit;
-	}
+  public itm: InterruptManager;
+  public em: EventManager;
+  public axm: AxisManager;
```

Comment:
I always thought this is quite a bad naming for public properties, how about changing it?
I think it should be done before releasing v3

Reply:
Also, is it fine to change the type names like IInputType, IInputTypeObserver to InputType and InputTypeObserver? There was an discussion before and I think it's time to change these names.